woocommerce/woocommerce
# Update the onboarding profile API validation for `business_extensions` parameter

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38766.

Update the onboarding profile API validation for `business_extensions` parameter to accept an array of strings.

While predefining a list of allowed values gives us full control over what values are accepted by the API, I think it's unnecessary to do that. In the front end, we have also defined it as the `string[]` type.

https://github.com/woocommerce/woocommerce/blob/32282a00ce839c773c110368824fcd899c78e616/packages/js/data/src/onboarding/types.ts#L128



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site
2. Open dev tool > network tab
3. Complete Core Profiler
4. Confirm `/wp-json/wc-admin/onboarding/profile` API POST returns 200



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
